appcli/templates/installer.j2

```diff
@@ -85,8 +97,27 @@ function print_error_and_exit()
     exit 1
 }
 
+function args()
+{
+    options=$(getopt --long auto-configure --long no-auto-configure -- "$@")
+    [ $? -eq 0 ] || {
+        echo "Incorrect option provided"
+        exit 1
+    }
+    eval set -- "$options"
+    while true; do
+        case "$1" in
+        --auto-configure) AUTO_CONFIGURE=true ;;
+        --no-auto-configure) AUTO_CONFIGURE=false ;;
+        --) shift; break; ;;
+        esac
+        shift
+    done
+}
+
 # ------------------------------------------------------------------------------
 # LOGIC
 # ------------------------------------------------------------------------------
 
-main $@
+args $0 "$@"
```

Comment:
It's not obvious why we call it like this, warrants a comment. Also, it should just be called from within `main` will update.